external/model-preparation-algorithm/mpa_tasks/extensions/datasets/mpa_cls_dataset.py

```diff
@@ -49,10 +55,49 @@ def get_indices(self, *args):
         return get_cls_img_indices(self.labels, self.ote_dataset)
 
     def load_annotations(self):
-        for dataset_item in self.ote_dataset:
-            roi_label = dataset_item.get_roi_labels(self.labels)
-            label = [self.label_idx[lbl.id] for lbl in roi_label] if roi_label else [None]
-            self.gt_labels.append(label)
+        include_empty = self.empty_label in self.labels
+        for i, _ in enumerate(self.ote_dataset):
+            class_indices = []
+            item_labels = self.ote_dataset[i].get_roi_labels(self.labels, include_empty=include_empty)
+            ignored_labels = self.ote_dataset[i].ignored_labels
+            if item_labels:
```

Comment:
Can't we separate the codes inside as a function?

Reply:
Okay, I'll separate this function on each dataset.